backbone.js

```diff
@@ -1202,6 +1217,69 @@
 
   });
 
+  // Defining an @@iterator method implements JavaScript's Iterable protocol.
+  // In modern ES2015 browsers, this value is found at Symbol.iterator.
+  /* global Symbol */
+  var $$iterator = typeof Symbol === 'function' && Symbol.iterator;
+  if ($$iterator) {
+    Collection.prototype[$$iterator] = Collection.prototype.values;
+  }
+
+  // CollectionIterator
+  // ------------------
+
+  // A CollectionIterator implements JavaScript's Iterator protocol, allowing the
+  // use of `for of` loops in modern browsers and interoperation between
+  // Backbone.Collection and other JavaScript functions and third-party libraries
+  // which can operate on Iterables.
+  var CollectionIterator = function(collection, kind) {
+    this._collection = collection;
+    this._kind = kind;
+    this._index = 0;
+  };
+
+  var KIND_VALUE = 1;
+  var KIND_KEY = 2;
+  var KIND_KEY_VALUE = 3;
```

Comment:
Yep, I prefer this for sure. Mind renaming these `ITERATOR_VALUEs`, `ITERATOR_KEYS`, `ITERATOR_KEYVALUES`